Gems/Profiler/Code/Include/Profiler/ImGuiTreemap.h

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include <AzCore/Name/Name.h>
+#include <AzCore/std/containers/vector.h>
+#include <AzCore/Interface/Interface.h>
+#include <imgui/imgui.h>
+
+namespace Profiler
+{
+    // TreemapNodes support arbitrary child counts and nesting levels. Each node will show up in the final visualization with an

```

Comment:
Really happy to see documentation in headers, we should though ideally follow the style code for these comments:  https://github.com/o3de/sig-core/blob/main/governance/API-Ref-Guidelines-Update.md

For example most of your class comments will be lost as they are regular comments.

Reply:
Thanks I read the style guide and added more documentation on top of what I had here. I wish I knew this doc existed before!